### ISSUE TITLE
Julia <=1.8.5 fails to compile with GCC 13

### ISSUE THREAD

**Issue by xgqt** [JuliaLang/julia#49581]
The blame is on the very outdated LLVM 13 version that Julia 1.8.5 bundles.

Please see the Gentoo bug report for more info: https://bugs.gentoo.org/895710

```
In file included from /var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/lib/Support/Signals.cpp:14:
/var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/include/llvm/Support/Signals.h:119:8: error: variable or field 'CleanupOnSignal' declared void
  119 |   void CleanupOnSignal(uintptr_t Context);
      |        ^~~~~~~~~~~~~~~
/var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/include/llvm/Support/Signals.h:119:24: error: 'uintptr_t' was not declared in this scope
  119 |   void CleanupOnSignal(uintptr_t Context);
      |                        ^~~~~~~~~
/var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/include/llvm/Support/Signals.h:18:1: note: 'uintptr_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   17 | #include <string>
  +++ |+#include <cstdint>
   18 | 
```
and
```
In file included from /var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/lib/Support/Signals.cpp:251:
/var/tmp/portage/dev-lang/julia-1.8.3-r4/work/julia-1.8.3/deps/srccache/llvm-julia-13.0.1-3/llvm/lib/Support/Unix/Signals.inc:348:44: error: 'void llvm::sys::CleanupOnSignal(uintptr_t)' should have been declared inside 'llvm::sys'
  348 | void sys::CleanupOnSignal(uintptr_t Context) {
      |                                            ^
```

**Comment by inkydragon** [comment]
Julia's CI is still using GCC9 on most platforms, except for using GCC 12 on Windows.

https://github.com/JuliaCI/rootfs-images/blob/939120f806e3e5f39d1a43e29ebceea052b6d0b7/linux/package_linux.jl#L62
```sh
    curl -fL $(repo_release_url)/GCC.v9.1.0.$(host_triplet)-$(cross_tags).tar.gz | tar zx
```

**Comment by vchuravy** [comment]
> very outdated LLVM 13

Released in Feb 2022, Julia 1.8 came out in August 2022.

Julia versions and LLVM versions are tightly matched so while it might seem like LLVM is outdated it is like saying that Clang 16 should be able to use LLVM 13.

**Comment by xgqt** [comment]
This bug is already fixed on master by a fellow (retired) Gentoo dev :) See: https://github.com/JuliaLang/llvm-project/commit/ff1681ddb303223973653f7f5f3f3435b48a1983
I do want to ask however: Why only GCC 9?
